audio-echo/app/src/main/java/com/google/sample/echo/MainActivity.java

```diff
@@ -80,7 +89,16 @@ public boolean onOptionsItemSelected(MenuItem item) {
     }
 
     public void startEcho(View view) {
-        status_view.setText("StartCapture Button Clicked\n");
+        int permissionStatus = ActivityCompat.checkSelfPermission(this,
```

Comment:
No need to store this in a variable if you are only checking it again in the next line.